### DISCUSSION TITLE
Issues with SWIG after Python import

### DISCUSSION THREAD

**Discussion by Keith-A-Madison** [NanoComp/meep#2367] [Q&A] [answered]
I've just configured MEEP on a fresh Ubuntu install using `apt-get install python3-meep h5utils`, as described in the MEEP docs. After importing MEEP, I'm met with the error:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3/dist-packages/meep/__init__.py", line 2600, in <module>
    class dft_ldos(object):
  File "/usr/lib/python3/dist-packages/meep/__init__.py", line 2639, in dft_ldos
    __swig_getmethods__["freq_min"] = get_omega_min
NameError: name '__swig_getmethods__' is not defined
```

I've tried installing SWIG with `pip install swig`, and am able to import it without error. I've tried importing SWIG prior to importing MEEP, to no avail. It's been a while since I've had to install MEEP, and I'm probably missing something obvious here, but shouldn't the aforementioned installation command be all that's needed? I see only one relevant previous discussion, which seems to indicate that this "issue" was fixed a while back with a previous version.

**Comment by stevengj** [comment]
Sounds similar to #1436 — are you using a really old version of Meep?

**Comment by smartalecH** [comment, marked as answer]
Linux distro binaries shipped by their own package managers (eg `apt`, `dnf`) aren't actively maintained. 

The recommended way to install meep is via conda, as those binaries _are_ actively maintained. 

https://meep.readthedocs.io/en/latest/Installation/#conda-packages
